snuba/settings/__init__.py

```diff
@@ -118,6 +134,25 @@
 REDIS_INIT_MAX_RETRIES = 3
 REDIS_REINITIALIZE_STEPS = 10
 
+T = TypeVar("T")
+
+
+class RedisClusters(TypedDict):
+    cache: RedisClusterConfig | None
+    rate_limiter: RedisClusterConfig | None
+    subscription_store: RedisClusterConfig | None
+    replacements_store: RedisClusterConfig | None
+    misc: RedisClusterConfig | None
+
+
+REDIS_CLUSTERS: RedisClusters = {
+    "cache": None,
+    "rate_limiter": None,
+    "subscription_store": None,
+    "replacements_store": None,
+    "misc": None,
```

Comment:
Why not removing this `misc` ? 
We can always map two logical clusters to the same physical cluster. 
Not having a `misc` will discourage people from using it for all new use cases and will remove the cognitive load of making a decision whether we need a new logical cluster for a new use case or not. 
Then a lot of small use cases will be mapped to the same physical cluster in the settings.

Reply:
i split up the usecase now. i was a bit hesitant because i didn't pay close attention there as to whether all those usecases in `snuba.state` can actually be split safely. ptal